src/main/java/filter/expression/Qualifier.java

```diff
@@ -717,6 +714,61 @@ private boolean milestoneSatisfies(IModel model, TurboIssue issue) {
         String contents = content.get().toLowerCase();
         String title = milestone.get().getTitle().toLowerCase();
 
+        List<TurboMilestone> allMilestones = model.getMilestones();
+        Collections.sort(allMilestones, new TurboMilestoneDueDateComparator());
+
+        int currentIndex = -1;
+
+        for (TurboMilestone checker : allMilestones) {
+            // current milestone must be open
+            if (checker.isOpen()) {
+                // if milestone is overdue but have no open issue, ignore
+                if (checker.getDueDate().isPresent() &&
+                        checker.getDueDate().get().isBefore(LocalDate.now()) &&
+                        checker.getOpenIssues() == 0) {
+                    continue;
+                }
+                currentIndex = allMilestones.indexOf(checker);
+                break;
+            }
+        }
+
+        // if no open milestone, set current as last milestone
+        if (currentIndex == -1 && allMilestones.size() > 0) {
+            currentIndex = allMilestones.size() - 1;
+        }
+
+        String alias = contents;
+
+        if (alias.equalsIgnoreCase("current") ||
+                alias.equalsIgnoreCase("curr")) {
+            contents = allMilestones.get(currentIndex).getTitle();
+        } else if (alias.equalsIgnoreCase("previous") ||
+                alias.equalsIgnoreCase("prev")) {
+            if (currentIndex - 1 >= 0) {
+                contents = allMilestones.get(currentIndex - 1).getTitle();
+            }
+        } else if (alias.equalsIgnoreCase("next")) {
+            if (currentIndex + 1 < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1).getTitle();
+            }
+        } else if (alias.startsWith("previous-") ||
+                alias.startsWith("prev-")) {
+            int offsetPos = alias.indexOf('-') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex - 1 - offset >= 0) {
+                contents = allMilestones.get(currentIndex - 1 - offset).getTitle();
+            }
+        } else if (alias.startsWith("next+")) {
+            int offsetPos = alias.indexOf('+') + 1;
+            int offset = Integer.parseInt(alias.substring(offsetPos).trim());
+            if (currentIndex + 1 + offset < allMilestones.size()) {
+                contents = allMilestones.get(currentIndex + 1 + offset).getTitle();
+            }
+        }
+
+        contents = contents.toLowerCase();
+
```

Comment:
@dariusf sorry, pressed wrongly. So in my current implementation, I replace the alias with a milestone name and just let it go to normal milestone search. It poses a problem since the milestone search is by `contains`, e.g. `current-10` now returns `V3.1` but it will include issues with milestone `V3.10` as well. Should I ignore this and post this as another issue (or searching milestone) or should alias use be strict match instead of `contains`?